server/async_functions.py

```diff
@@ -0,0 +1,23 @@
+"""
+Some helper functions for common async tasks.
+"""
+import asyncio
+from typing import List
+
+
+async def gather_without_exceptions(
+    tasks: List[asyncio.Task],
+    *exc_classes: type,
+) -> None:
+    """
+    Call gather on a list of tasks, raising the first exception that dosen't
+    match any of the specified exception classes.
+    """
+    results = await asyncio.gather(*tasks, return_exceptions=True)
+    for result in results:
+        if isinstance(result, Exception):
+            # Check if this exception is an instance (maybe subclass) that
+            # should be ignored
+            for exc_type in exc_classes:
+                if not isinstance(result, exc_type):
+                    raise result
```

Comment:
I think, we should return `results`, if it's possible, even if it's not needed now. Maybe we would like process/log that results..